shadowsocks/stream.go

```diff
@@ -21,11 +21,17 @@ import (
 	"fmt"
 	"io"
 	"sync"
+
+	"github.com/Jigsaw-Code/outline-ss-server/slicepool"
 )
 
 // payloadSizeMask is the maximum size of payload in bytes.
 const payloadSizeMask = 0x3FFF // 16*1024 - 1
 
+// Buffer pool used for encrypting and decrypting Shadowsocks streams.
+// The largest buffer we could need is for encrypting a max-length payload.
+var ssPool = slicepool.MakePool(payloadSizeMask + maxTagSize())
```

Comment:
Perhaps name it tcpPool, so we don't use it from the UDP code?

Reply:
Renamed to `readBufPool`.